### PR TITLE
Fixes #18145 - add RHSM log to katello-debug

### PR DESCRIPTION
**Pull request by chris1984**

### CODE REVIEW

**Comment by mccun934** [comment]
Will this error out if the file doesn't exist, say on systems that aren't using rhsm?

**Comment by chris1984** [comment]
[root@dhcp-xx ~]# service rhsmcertd stop - Stopping RHSM from logging
Redirecting to /bin/systemctl stop  rhsmcertd.service
[root@dhcp-xx ~]# rm -rf /var/log/rhsm/rhsm.log - Removing the log
[root@dhcp-xx ~]# vi /usr/share/foreman/script/foreman-debug.d/katello-debug.sh adding the PR
[root@dhcp-xx ~]# ll /var/log/rhsm/ - verifying the log file is missing
-rw-r--r--. 1 root root 521 Jan 18 16:07 rhsmcertd.log

[root@dhcp-xx ~]# foreman-debug 
Processing... (takes a while)Exporting tasks, this may take a few minutes.
Copying entire file: /tmp/task-export-1484773753.tar.gz

 HOSTNAME: dhcp-xx.example.com
       OS: redhat
  RELEASE: Red Hat Enterprise Linux Server release 7.2 (Maipo)
  FOREMAN: 1.11.0.61
     RUBY: ruby 2.0.0p648 (2015-12-16) [x86_64-linux]
   PUPPET: 3.8.6
  DENIALS: 0
A debug file has been created: /tmp/foreman-debug-xMbA6.tar.xz (392452 bytes)
To upload a tarball to our secure site, please use the -u option.

[root@dhcp-xx foreman-debug-xMbA6]# cd var/log/
[root@dhcp-xx log]# ls
audit  candlepin  foreman  foreman-installer  foreman-proxy  httpd  messages  mongodb  squid  tomcat

@mccun934 looks good from a test here

File: katello/katello-debug.sh [hunk 1]
```diff
@@ -57,6 +57,9 @@ add_files /var/log/tomcat*/manager*.log*
 add_files /etc/candlepin/candlepin.conf
 add_files /etc/tomcat*/server.xml
 
+# RHSM
+add_files /var/log/rhsm/rhsm.log
```
Comment (sean797) [inline katello/katello-debug.sh]: Probably should be `/var/log/rhsm/rhsm*.log` or `/var/log/rhsm/*.log`. Some people rotate these :-)

**Comment by chris1984** [comment]
Testing:

Katello 3.2 on CENTOS7:

Missing RHSM log dir/files

[root@centos7-katello-3-2 foreman-debug.d]# ls /var/log/
anaconda  boot.log  candlepin  dmesg    foreman-installer  grubby  koan     maillog   mongodb              ppp   puppet     secure   squid     tomcat  wtmp
audit     btmp      cron       foreman  foreman-proxy      httpd   lastlog  messages  passenger-analytics  pulp  qdrouterd  spooler  tallylog  tuned   yum.log

Run of foreman-debug:

[root@centos7-katello-3-2 foreman-debug.d]# foreman-debug 
Processing... (takes a while)Exporting tasks, this may take a few minutes.
Copying entire file: /tmp/task-export-1484927190.tar.gz


 HOSTNAME: centos7-katello-3-2.example.com
       OS: redhat
  RELEASE: CentOS Linux release 7.3.1611 (Core) 
  FOREMAN: 1.14.0-develop
     RUBY: ruby 2.0.0p648 (2015-12-16) [x86_64-linux]
   PUPPET: 3.8.7
  DENIALS: 86


A debug file has been created: /tmp/foreman-debug-9TdX4.tar.xz (483312 bytes)

To upload a tarball to our secure site, please use the -u option.

Untar of debug without any errors:

[root@centos7-katello-3-2 log]# ls
audit  candlepin  foreman  foreman-installer  foreman-proxy  httpd  messages  mongodb  squid  tomcat
[root@centos7-katello-3-2 log]# pwd
/tmp/foreman-debug-9TdX4/var/log

Downstream 6.2:

Showing RHSM log dir/files:

[root@satellite1 foreman-debug.d]# ls /var/log/rhsm/
rhsmcertd.log  rhsmcertd.log-20161225  rhsmcertd.log-20170102  rhsmcertd.log-20170108  rhsmcertd.log-20170116  rhsm.log  rhsm.log-20161225  rhsm.log-20170102  rhsm.log-20170108  rhsm.log-20170116

Run of foreman-debug:

[root@satellite1 tmp]# foreman-debug 
Processing... (takes a while)Exporting tasks, this may take a few minutes.
Copying entire file: /tmp/task-export-1484927616.tar.gz


 HOSTNAME: satellite1.released-el7.satellite.lab.eng.rdu2.redhat.com
       OS: redhat
  RELEASE: Red Hat Enterprise Linux Server release 7.3 (Maipo)
  FOREMAN: 1.11.0.61
     RUBY: ruby 2.0.0p648 (2015-12-16) [x86_64-linux]
   PUPPET: 3.8.6
  DENIALS: 178


A debug file has been created: /tmp/foreman-debug-u73O3.tar.xz (5967460 bytes)

To upload a tarball to our secure site, please use the -u option

Untar of debug without any errors:

[root@satellite1 rhsm]# ls
rhsmcertd.log  rhsmcertd.log-20161225  rhsmcertd.log-20170102  rhsmcertd.log-20170108  rhsmcertd.log-20170116  rhsm.log  rhsm.log-20161225  rhsm.log-20170102  rhsm.log-20170108  rhsm.log-20170116
[root@satellite1 rhsm]# pwd
/tmp/foreman-debug-ttD8C/var/log/rhsm

**Comment by chris1984** [comment]
@mccun934 @sean797 When you get a chance can I get a review again. Made the corrected changes and posted the output of my tests. I used /* since that is what we are using in the other log files ie:

add_files /var/log/foreman-installer/*

Also we get that info in the sosreport downstream and there is nothing personal in there. Upstream it should not be an issue since there is no RHSM.

**Review by sean797 — APPROVED** [review]
Looks good to me! Thanks @chris1984

**Review by ehelms — APPROVED** [review]

**Pull request merged**